pkg/cover/backend/dwarf.go

```diff
@@ -489,24 +520,16 @@ func readCoverPoints(target *targets.Target, info *symbolInfo, data []byte) ([2]
 		return pcs, fmt.Errorf("no __sanitizer_cov_trace_pc symbol in the object file")
 	}
 
-	// Loop that's checking each instruction for the current architectures call
-	// opcode. When found, it compares the call target address with those of the
-	// __sanitizer_cov_trace_* functions we previously collected. When found,
-	// we collect the pc as a coverage point.
-	arch := arches[target.Arch]
-	for i, opcode := range data {
-		if opcode != arch.opcodes[0] && opcode != arch.opcodes[1] {
-			continue
-		}
-		i -= arch.opcodeOffset
-		if i < 0 || i+arch.callLen > len(data) {
-			continue
+	i := 0
+	for {
+		arch := arches[target.Arch]
```

Comment:
Maybe 

```
arch, ok := arches[target.Arch]
if !ok {
  panic(fmt.Sprintf("readCoverPoints: unknown arch %s", target.Arch))
}
i := 0
for {
```

?

FWIW after this you may take a reference of `arch`, though I'm not sure it makes any difference here.

Reply:
You're right, I can copy arches[target.Arch] in the test as well, then it becomes possible to take a reference. Dunno if it hasn't become uglier though.